src/main/java/webdriver/browser/Browser.java

```diff
@@ -4,15 +4,18 @@
 import org.openqa.selenium.interactions.Actions;
 import org.openqa.selenium.support.ui.ExpectedConditions;
 import org.openqa.selenium.support.ui.WebDriverWait;
+import webdriver.driver.DriverManager;
 
 import java.util.Set;
 
 public class Browser {
     private static final int WAITING_TIME = 10;
     private WebDriver driver;
 
-    public Browser(WebDriver driver) {
-        this.driver = driver;
+    public Browser() {
+        if (driver == null) {
```

Comment:
What if driver !=null? It won't be initialized? That doesn't make sense. driver will always be null if you call a constructor cause it's not static.  That is not a singleton when you can create an instance of the object using its constructor